packages/vite/src/node/plugins/html.ts

```diff
@@ -317,33 +564,28 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
 
           // For asset references in index.html, also generate an import
           // statement for each - this will be handled by the asset plugin
-          const assetAttrs = assetAttrsConfig[node.tag]
-          if (assetAttrs) {
-            for (const p of node.props) {
-              if (
-                p.type === NodeTypes.ATTRIBUTE &&
-                p.value &&
-                assetAttrs.includes(p.name)
-              ) {
-                // assetsUrl may be encodeURI
-                const url = decodeURI(p.value.content)
-                if (!isExcludedUrl(url)) {
-                  if (node.tag === 'link' && isCSSRequest(url)) {
-                    // CSS references, convert to import
-                    js += `\nimport ${JSON.stringify(url)}`
-                    shouldRemove = true
-                  } else {
-                    assetUrls.push(p)
-                  }
-                } else if (checkPublicFile(url, config)) {
-                  s.overwrite(
-                    p.value.loc.start.offset,
-                    p.value.loc.end.offset,
-                    `"${config.base + url.slice(1)}"`,
-                    { contentOnly: true }
-                  )
-                }
+          const assetSourceAttrs = getHtmlAssetSourceAttributes(
+            node,
+            config.build.htmlAssetSources
+          )
+          for (const assetSourceAttr of assetSourceAttrs) {
+            // assetsUrl may be encodeURI
+            const url = decodeURI(assetSourceAttr.value!.content)
+            if (!isExcludedUrl(url)) {
+              if (node.tag === 'link' && isCSSRequest(url)) {
```

Comment:
Should prefetch / preload be loaded directly?

Reply:
Is this question unrelated to the PR, no? Could you explain a bit more?